packages/front-end/components/Settings/ClickToReveal.tsx

```diff
@@ -1,59 +1,111 @@
-import { ReactElement, ReactNode, useState } from "react";
-import { FaExclamationTriangle } from "react-icons/fa";
-import LoadingSpinner from "../LoadingSpinner";
+import { SecretApiKey } from "back-end/types/apikey";
+import clsx from "clsx";
+import { useState } from "react";
+import { FaExclamationTriangle, FaEye, FaEyeSlash } from "react-icons/fa";
+import { useAuth } from "../../services/auth";
 import Tooltip from "../Tooltip";
 
+export type SecretKey = {
+  [key: string]: string;
+};
 export interface Props {
-  valueWhenHidden: ReactNode;
-  getValue: () => Promise<string>;
-  children: (value: string) => ReactElement;
+  rowReverse?: boolean;
+  keyId: string;
+  currentCopiedString: string;
+  setCurrentCopiedString: (value: string) => void;
 }
 
-export default function ClickToReveal({
-  valueWhenHidden,
-  getValue,
-  children,
+export default function RevealHiddenKey({
```

Comment:
You changed the name of the component, but not the filename.  That's going to cause some IDE autocomplete issues.